### DISCUSSION TITLE
Implementing Thermal Effetcs

### DISCUSSION THREAD

**Discussion by TheIranCircle** [InteractiveComputerGraphics/PositionBasedDynamics#199] [General]
I'd like to try adding thermal effects such as thermal expansion or softening and possibly phase transformation. I have started adding  variables to the particle such as `std::vector<Real> m_temperature;`, but it would be really helpful to have a few pointer on where to start implementing the coupling like this? I am thinking to try implementing loose coupling first. Thanks

**Comment by janbender** [comment]
Do you implement this in PBD, XPBD or Shape Matching?
